docs/manual-EN/5.appendix/gremlin-ngql.md

```diff
@@ -0,0 +1,247 @@
+
+- Syntax is close to SQL, but not exactly the same (Easy to learn)
+- Expandable
+- Case insensitive
+- Support basic graph traverse
+- Support pattern match
+- Support aggregation
+- Support graph mutation
+- Support distributed transaction (future release)
+- Statement composition, but **NO** statement embedding (Easy to read)
+
+## Conceptual Comparisons
+
+Name               | Gremlin | nGQL          |
+-----              |---------|   -----       |
+vertex, node       | vertex  | vertex        |
+edge, relationship | edge    | edge          |
+vertex type        | label   | tag           |
+edge type          | label   | edge type     |
+
+In Gremlin and nGQL, vertices and edges are identified with unique identifiers. In **Nebula Graph**, you can either specify identifiers or generate automatically with the hash or uuid function.
+
+## Basic Graph Operations
+
+Name                   | Gremlin         | nGQL          |
+-----                  |---------        |   -----       |
+Create a new graph     | g = TinkerGraph.open().traversal() | CREATE SPACE gods |
+Show vertices' types   | g.V().label()   | SHOW TAGS |
+Insert a vertex with specified type | g.addV(String vertexLabel).property() | INSERT VERTEX <tag_name> (prop_name_list) VALUES vid:(prop_value_list) |
+Insert an edge with specified edge type | g.addE(String edgeLabel).from(v1).to(v2).property()| INSERT EDGE <edge_name> ( <prop_name_list> ) VALUES <src_vid> -> <dst_vid>: ( <prop_value_list> ) |
+Delete a vertex | g.V(\<vid>).drop() | DELETE VERTEX \<vid> |
+Delete an edge  | g.E(\<vid>).outE(\<type>).where(otherV().is(\<vid>))drop() | DELETE EDGE <edge_type> \<vid> -> \<vid> |
+Update a vertex property | g.V(\<vid>).property() | UPDATE VERTEX <vid> SET <update_columns> |
+Fetch vertices with ID | g.V(\<vid>) | FETCH PROP ON <tag_name> \<vid>|
+Fetch edges with ID    | g.E(<start_vid> >> <dst_vid>) | FETCH PROP ON <edge_name> <start_vid> -> <dst_vid> |
+Query a vertex along specified edge type | g.V(\<vid>).outE( \<edge>) | GO FROM \<vid> OVER  \<edge> |
+Query a vertex along specified edge type reversely | g.V(\<vid>).in( \<edge>) | GO FROM \<vid>  OVER \<edge> REVERSELY |
+Query N hops along a specified edge | g.V(\<vid>).repeat(out(\<edge>)).times(N) | GO N STEPS FROM \<vid> OVER \<edge> |
+Find path between two vertices | g.V(\<vid>).repeat(out()).until(\<vid>).path() | FIND ALL PATH FROM \<vid> TO \<vid> OVER * |
+
+## Example Queries
+
+The examples in this section make extensive use of the toy graph distributed with [Janus Graph](https://janusgraph.org/) called [_The Graphs of Gods_](https://docs.janusgraph.org/#getting-started). This graph is diagrammed below. The abstract data model is known as a [Property Graph Model](https://github.com/vesoft-inc/nebula/blob/master/docs/manual-EN/1.overview/1.concepts/1.data-model.md) and this particular instance describes the relationships between the beings and places of the Roman pantheon.
```

Comment:
The examples in this section make extensive use of the diagram

Reply:
i think it's ok here

Reply:
what is a toy graph?